File: erp_api/ERP.API/Domain/Entities/LeaveType.cs
namespace ERP.API.Domain.Entities;

public class LeaveType
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public int DefaultDays { get; set; }
    public bool IsPaid { get; set; }
    public bool RequiresApproval { get; set; } = true;
    public string? IconCode { get; set; }
    public string? ColorCode { get; set; }
    public int MaxConsecutiveDays { get; set; }
    public bool IsActive { get; set; } = true;
}

File: erp_flutter/lib/core/models/leave_type.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'leave_type.freezed.dart';
part 'leave_type.g.dart';

@freezed
abstract class LeaveType with _$LeaveType {
  const LeaveType._();

  factory LeaveType({
    required String id,
    required String name,
    String? description,
    required int defaultDays,
    required bool isPaid,
    required bool requiresApproval,
    String? iconCode,
    String? colorCode,
    required int maxConsecutiveDays,
    required bool isActive,
  }) = _LeaveType;

  factory LeaveType.fromJson(Map<String, dynamic> json) =>
      _$LeaveTypeFromJson(json);
}
